tests/Unit/DataStructures/Tensor/Expressions/Test_MixedOperations.cpp

```diff
@@ -129,7 +129,7 @@ void test_case1(const DataType& used_for_size,
   result_tensor_type actual_result_tensor_returned =
       TensorExpressions::evaluate<ti_a>(R(ti_a, ti_b) * S(ti_B) + G(ti_a) -
                                         H(ti_b, ti_a, ti_B) * T());
-  result_tensor_type actual_result_tensor_filled{};
+  result_tensor_type actual_result_tensor_filled(used_for_size);
```

Comment:
Why is this necessary now? I may have missed this, but is the general policy that the result buffer must be sized correctly on entry, or will `TE::evaluate` resize it? Is this documented?

Reply:
I think this was a vestige of a time during the design process when this necessary, but it no longer is, so I've removed it. TEs should work whether the input LHS tensor has been sized or not because it uses assignment like `lhs_component = get(result of some subtree)`, which should assign the LHS component using a sized RHS `Tensor` component

Reply:
Now that I think about it, I think the `get_used_for_size` functions I've added to each derived TE class probably isn't needed anymore for the same reason - not unless there is an expected performance difference between the following two cases:

(a)
```
tnsr::i<DataVector> R(used_for_size);
get<0>(R) = ...;
get<1>(R) = ...;
get<2>(R) = ...;
```

(b)
```
tnsr::i<DataVector> R{};
get<0>(R) = ...;
get<1>(R) = ...;
get<2>(R) = ...;
```

Do either you or @nilsdeppe know if (a) or (b) is generally preferred? If not, it's something I could look into

Reply:
Addressed by https://github.com/sxs-collaboration/spectre/pull/3842/commits/b1566dabe4805a70c10e5a17c698a07774d3fbdb